dbms/src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -2791,22 +3166,21 @@ String StorageReplicatedMergeTree::getFakePartNameCoveringAllPartsInPartition(
 
     {
         auto zookeeper = getZooKeeper();
-        AbandonableLockInZooKeeper block_number_lock = allocateBlockNumber(partition_id, zookeeper);
-        right = block_number_lock.getNumber();
-        block_number_lock.unlock();
+        auto block_number_lock = allocateBlockNumber(partition_id, zookeeper);
+        right = block_number_lock->getNumber();
+        block_number_lock->unlock();
     }
 
     /// Empty partition.
     if (right == 0)
-        return {};
+        return false;
 
     --right;
 
-    if (out_min_block)
-        *out_min_block = left;
-    if (out_max_block)
-        *out_max_block = right;
-    return getFakePartNameCoveringPartRange(data.format_version, partition_id, left, right);
+    /// Artificial high level is choosen, to make this part "covering" all parts inside.
+    static constexpr UInt32 level = 999999999;
```

Comment:
`std::numeric_limits::max`

Reply:
By some reason, this value has been already used for a long period of time.